### ISSUE TITLE
Oxyacetylene torch - Use as a lighter?

### ISSUE THREAD

**Issue by SeanMirrsen** [CleverRaven/Cataclysm-DDA#14453]
In a case of massive utility overkill, my character is stuck on a cold night inside a hardware store in the middle of nowhere, and has no way to make light, or a fire, or even sufficiently warm clothing. But there is an oxyacetylene torch on the store shelves. As far as I understood, the torch is self-lighting, as I don't recall needing a lighter to start it up.
However, the only action the torch seems to be usable for, is cutting metal. Attempting to use it tells you to get welding goggles, even if you want to, say, set fire to a bush with it. It melts metal, surely it can light some wood?

**Comment by sparr** [comment]
This should probably work. Hell, you should be able to start a fire with a powered welder if you want, too.

**Comment by Coolthulhu** [comment]
It would be quite easy to add it.
Also, extended firestarter could be added to soldering iron and welder.

**Comment by Malkeus** [comment]
You'd probably want to have a couple pieces of scrap metal for the welder.
Arcing off the terminals directly would degrade (read melt) them fairly
quickly. Fires happen accidentally pretty frequently tbh, at least when I'm
welding. I keep a spray bottle of water and a fire extinguisher handy.

**Comment by chaosvolt** [comment]
Only issue is whether or not lighting a fire via an acetylene torch should use the same number of charges as cutting through a door. :v

**Comment by Rivet-the-Zombie** [comment]
>Only issue is whether or not lighting a fire via an acetylene torch should use the same number of charges as cutting through a door. :v

It should take way less; lighting a fire with an oxy torch takes less time than it takes to pass the flame over your combustibles, whereas cutting metal requires a bit of preheating.

**Comment by Malkeus** [comment]
The flame is adjustable as well. You can get a small area very very hot, or
create a flame similar to a lighter, but over a foot long.

**Comment by Rivet-the-Zombie** [comment]
And turn your entire cigarette into ashes when you try to light it with one.

That stuff only looks cool in the movies, kids.

**Comment by Malkeus** [comment]
Nonsense, the trick is holding your cigarette NEAR the flames, not in them.
You can light a cigarette off the hot metal edges of whatever you just cut
as well. That's actually the safer method, less chance of a stray wind
searing your eyebrow off.

**Comment by Rivet-the-Zombie** [comment]
I've never had any luck lighting one off the edge of the flame either. Seems like if it's close enough to light it, then it's close enough to turn the first inch and a half into ash.

**Comment by Malkeus** [comment]
There's a sweet spot at the back of the flame, behind the nozzle.
Especially if it's adjusted for cutting.

**Comment by chaosvolt** [comment]
This leads right into the issue that we need a way for different use_actions in a given tool to allow differing charges_per_use. Same reason toolboxes have that "using the hammer devours duct tape" issue. o3o

**Comment by kevingranade** [comment]
If the only actions being contemplated are "cut through door or metal bars"
and "light a cigarette", it's not a critical need.

**Comment by chaosvolt** [comment]
What isn't a critical need, the idea of shifting "charges per use" to the use actions?

If so, you weren't paying attention when I mentioned the toolbox problem. >_>

**Comment by SeanMirrsen** [comment]
The action being contemplated is "start a fire to stop your legs from freezing off", so for a given, very small, subset of characters who have found an acetylene torch or welder but don't have a matchbook or a lighter and are in dire need of a fire, it is rather a critical need.

**Comment by kevingranade** [comment]
If you're referencing some random other issue, yes I'll tend to ignore it
because it's irrelevant to this issue.

**Comment by chaosvolt** [comment]
The suggestion that I've made regarding that old issue AND this one is relevant.

If a lighter function is to be added to the acetylene torch, it wouldn't make sense for lighting something to use the same number of charges as cutting through a door.

Likewise, the change needed to make cutting things with the toolbox not consume duct tape could be implemented by the same change that would make this idea sensible.

Put simply, I was suggesting something that would be useful for this issue's suggestion, then providing ANOTHER example of an issue that could be resolved by the same method.

So naturally you immediately dismissed my suggestion as only being relevant to THIS suggestion because you weren't paying attention. >.>

**Comment by mugling** [comment]
Implementation would be trivial. Discussion could continue in any submitted PR

**Comment by chaosvolt** [comment]
Implementation of this still hinges on shifting "charges per use" from the item to the actions. Is there an issue to suggest this?

**Comment by mugling** [comment]
Actually no, please disregard the trivial part of the above.

**Comment by chaosvolt** [comment]
That's kinda what the entire ancient discussion on here was about. Well, half of it was laying out the code that this would need to behave sanely, while the other half was discussing whether it's a sane thing to do in real life.

I recall mentioning that the idea of a "charges per use" shift is useful for other buggy multi-function items, way back when. Admittedly, you aren't the first to miss that point, as @kevingranade did back when I first brought it up.

**Comment by mugling** [comment]
I closed this along with a number of other bugs that fulfilled the criteria of both triviality and lack of obvious progress

> it's not a critical need.

I see no disagreement with those criteria.

> Same reason toolboxes have that "using the hammer devours duct tape"

This was fixed some time ago

> Implementation would be trivial

It could now be done via extension of `firestarter_actor`. 

It's important to prune the outstanding issues once in a while so that we can check important ones aren't being left to rot. The purpose isn't really to resume discussion in them unless there is something new to add, which leaves us with this:

> Discussion could continue in any submitted PR

**Comment by chaosvolt** [comment]
I see. So the original problem behind this suggestion has been worked out by PRs that have since been added? If that's true, then very well then.

Granted, if this needed further discussion, you know bloody well I wouldn't be the one PRing it, and chances are I'd never find said PR to add said discussion if someone else did so.

**Comment by mugling** [comment]
> So the original problem behind this suggestion has been worked out by PRs that have since been added?

The original issue *"Oxyacetylene torch - Use as a lighter?"* was declared to be of only trivial importance and as no progress was made in 8 months the issue was closed

**Comment by chaosvolt** [comment]
That would be the reason why this idea never got very far in discussion, not the problem that I was referring to. As in, what would've prevented any PR adding this suggestion to work as intended, the whole "lighter function would use same amount of charges" issue.

You said earlier that `firestarter_actor` can now be used to work around this issue. If that's true, then that answers the question I was asking.
